fix(i18n): Address internationalization snags and inconsistencies

shivambisen opened the pull request:
- Removed unnecessary Carbon icon from UI
- Loaded localized markdown file for DE locale
- Fixed layout issues caused by longer German translations
- Ensured fallback to English when feature flag is off
- Corrected pluralization and label mismatches ("Benutzeren" → "Benutzer", etc.)
- Applied proper translations in pagination controls via message catalog
- Fixed MISSING_MESSAGE error on Test Run row click
- Corrected "Access Denied" header mistakenly shown on Test Runs page

Michael-Baylis commented:
rerun tests

shivambisen commented:
missing Test Run name:

![WhatsApp Image 2025-06-21 at 12 54 52 PM (1)](https://github.com/user-attachments/assets/0272d414-0f4f-41b9-ac65-d416bf5cda03)

translations in pagination :

![WhatsApp Image 2025-06-21 at 12 54 52 PM](https://github.com/user-attachments/assets/5f01456b-cc2d-4b97-84b9-704a492471bb)

fixed "Access Denied" header mistakenly shown on Test Runs page: 

![WhatsApp Image 2025-06-21 at 12 54 51 PM (1)](https://github.com/user-attachments/assets/9a2ee16e-1c58-4398-9bf4-3205fc0b0c28)

delete Button layout in my setting page : 

![WhatsApp Image 2025-06-21 at 12 54 51 PM](https://github.com/user-attachments/assets/62df8c69-30b4-4def-96c7-5c3e6e0f0439)

localized markdown file for DE locale:

![WhatsApp Image 2025-06-21 at 12 54 46 PM](https://github.com/user-attachments/assets/1de2c152-20c0-4deb-ad44-8d6a6e158670)

eamansour commented on galasa-ui/messages/de.json:
```diff
@@ -4,7 +4,7 @@
   "Breadcrumb": {
     "home": "Startseite",
-    "users": "Benutzer",
+    "users": "Benutzeren",
```
I know this was raised by @techcobweb, but I'm certain "Benutzeren" isn't a word. "Benutzer" can be singular and plural - e.g. "der Benutzer" (singular), "die Benutzer" (plural).

We can keep it as it was previously - same applies throughout this PR.

eamansour commented on galasa-ui/messages/de.json:
```diff
@@ -191,14 +193,14 @@
   },
   "TestRunDetails": {
-    "title": "Testlauf:",
+    "title": "Testlauf: ({runName})",
```
Can we remove the surrounding `()` from the test run name?

eamansour commented on galasa-ui/messages/en.json:
```diff
@@ -194,7 +193,7 @@
   },
   "TestRunDetails": {
-    "title": "Test Run:",
+    "title": "Test Run:({runName})",
```
Can we remove the surrounding `()` from the test run name and add a space between "Test Run:" and the name?

eamansour commented on galasa-ui/src/app/page.tsx:
```diff
@@ -52,21 +55,28 @@ export default function HomePage() {
+
+    const fileToRead = fs.existsSync(localizedFilePath)
+      ? localizedFilePath
+      : fallbackFilePath;
```
This looks good - can we add a unit test to check that the correct file is selected based on the locale provided?

eamansour commented on galasa-ui/src/app/page.tsx:
```diff
@@ -75,9 +85,15 @@ export default function HomePage() {
+    markdownContentPromise = fetchHomePageContentFromCps().catch(() =>
+      fetchDefaultMarkdownContent( locale ),
+    );}else{
+    markdownContentPromise = fetchDefaultMarkdownContent(locale);
```
This surrounding `if` block isn't necessary. A user might have custom homepage content to show stored in the Galasa service's configuration properties store (CPS), so we should always try to get content from that before falling back to a default if there is no custom homepage set.

shivambisen replied:
Thanks @eamansour I added the if (locale === "en") block because , CPS only contains the homepage markdown in English. So if the user selects a different locale (like "de"), we skip the CPS call and directly load the localized markdown file from the public (e.g., home-contents.de.md).

If we always call CPS first, users selecting a non-English language will still get the English content.

eamansour replied:
Thanks for the explanation.

This behaviour is because we have a CPS property within the ecosystem1 service called `service.welcome.markdown`, which is currently set to:
```
# Welcome to your Galasa service
        To Make the most of your Galasa experience, please refer to the [Galasa documentation](https://galasa.dev/)
        ![image](https://raw.githubusercontent.com/galasa-dev/webui/ee5cf521ebb392079b3d672c78ad2da128d8fef9/galasa-ui/public/static/homeGraphic.svg)
```

This value can be customised to contain any markdown content we want. So, if some German users had their own Galasa service deployed and they set this CPS property to include content in German, then that would be rendered.

This `if` block will unfortunately break this functionality, so only users using the `en` locale would be able to customise their homepage using the CPS property value.

Hope that makes sense!

eamansour commented on galasa-ui/src/tests/index.test.tsx:
```diff
@@ -60,9 +60,10 @@ test('renders Galasa header', () => {
 
 test('renders Galasa Ecosystem homepage', async () => {
   const homePage = await act(async () => {
+    const layout = await HomePage();
```
Very minor - but `layout` isn't quite accurate here, maybe `homePageComponent`?

eamansour reviewed (commented):
These changes are looking good! Just a few improvements that can be made. Let me know if any don't make sense!

shivambisen committed:
chore: fix i18n entries, adjust homepage markdown logic, and add unit tests

- Implemented minor fixes
- Add unit tests verifying getMarkdownFilePath chooses correct file based on locale:
  • When localized file exists → path contains “home-contents.{locale}.md”
  • When missing → fallback to “home-contents.md”

Signed-off-by: ShivamBisen <shivambisen16@kgpian.iitkgp.ac.in>
(cherry picked from commit e5ab9c6fcb3bedede4ad8280b3d4be0679c944a5)

shivambisen commented:
@eamansour I've made the suggested changes 
Let me know if anything else needs tweaking :)

eamansour commented on galasa-ui/messages/de.json:
```diff
@@ -13,7 +13,7 @@
   },
   "PageHeader": {
-    "users": "Benutzer",
+    "users": "Benutzeren",
```
This "Benutzeren" still needs changing back

eamansour commented on galasa-ui/src/utils/markdown.ts:
```diff
@@ -0,0 +1,12 @@
+// src/utils/markdown.ts
```
This file is missing Galasa's copyright statement

eamansour reviewed (commented):
Two very minor comments, but the changes look good, so I'll be happy to approve once they've been resolved and the new merge conflicts have been resolved.

shivambisen commented:
@eamansour Thanks for the review! I've addressed both comments and resolved the merge conflicts .

eamansour commented on galasa-ui/messages/en.json:
```diff
@@ -228,6 +227,8 @@
     "result": "Result",
     "noTestRunsFound": "No test runs were found for the selected timeframe",
     "loading": "Loading test results...",
+    "noData": "No test runs found in the last 24 hours.",
```
We've just delivered a change that allows users to search for runs in a custom timeframe, so it no longer defaults to 24 hours.

It looks like there's a translation above called `noTestRunsFound` which we can use instead of `noData` here.

The same applies for the de messages

eamansour commented on galasa-ui/messages/en.json:
```diff
@@ -228,6 +227,8 @@
     "noTestRunsFound": "No test runs were found for the selected timeframe",
     "loading": "Loading test results...",
+    "noData": "No test runs found in the last 24 hours.",
+    "isloading":"Loading test results...",
```
Looks like we now have a `loading` translation and an `isloading` translation, both with the same message. It would be great if we could just have one, `isLoading` would be fine!

Again, same applies for the de messages :)

eamansour reviewed (commented):
Sorry for the hassle, just spotted two more minor issues now that the merge conflicts have been resolved! These will hopefully be my last comments :)

shivambisen commented:
No worries at all ,thanks for the careful review! I've addressed the issues. Please let me know if anything else needs updating

eamansour reviewed (approved):
Looks great! Thanks for making those changes - approved.

eamansour commented:
@shivambisen it seems like your last commit might not have been signed/verified, which is blocking us from merging this PR. Are you able to perhaps re-commit your last changes? Thanks!

shivambisen committed:
fix(i18n): Address internationalization snags and inconsistencies

- Removed unnecessary Carbon icon from UI
- Loaded localized markdown file for DE locale
- Fixed layout issues caused by longer German translations
- Ensured fallback to English when feature flag is off
- Corrected pluralization and label mismatches ("Benutzeren" → "Benutzer", etc.)
- Applied proper translations in pagination controls via message catalog
- Fixed MISSING_MESSAGE error on Test Run row click
- Corrected "Access Denied" header mistakenly shown on Test Runs page

Signed-off-by: ShivamBisen <shivambisen16@kgpian.iitkgp.ac.in>

shivambisen committed:
fix:addressed issues,merge conflicts and signature issue

Signed-off-by: ShivamBisen <shivambisen16@kgpian.iitkgp.ac.in>

shivambisen commented:
Hello @eamansour  Sorry for the trouble. one of my previous commits wasn’t GPG-signed. I’ve re-signed the relevant commits and force-pushed the changes. It should be good to go now. And again, sorry for the problem

eamansour commented:
No problem at all!

eamansour commented:
@shivambisen the changes have deployed on https://galasa-ecosystem1.galasa.dev/ but seem to be causing client-side exceptions, with this error:
```
 ⨯ Error: Server Functions cannot be called during initial render. This would create a fetch waterfall. Try to use a Server Component to pass data to Client Components instead.
    at PageHeaderMenu (./src/components/headers/PageHeaderMenu.tsx:44:69)
digest: "3803599244"
 GET / 500 in 109ms
```

The issue seems to be coming from this block of code in PageHeaderMenu:
```ts
  if(!isFeatureEnabled(FEATURE_FLAGS.INTERNATIONALIZATION)) {
    setUserLocale("en");
  }
```

Are you able to investigate this?


Edit: I've just opened a PR to fix this issue https://github.com/galasa-dev/webui/pull/131

shivambisen commented:
@eamansour  Thanks a lot for flagging this and fixing it quickly! 
Really sorry ,I completely forgot to  use useEffect for conditional rendering .
I’ll make sure to be more careful about this in the future. I believe I handled it correctly in the Theme Support PR, but must’ve missed it here. 

but i really want to learn what’s the best way to catch these kinds of issues during local development? Sometimes they don’t show up clearly when testing locally.

eamansour commented:
No worries @shivambisen! Issues like this would be best caught using unit tests (e.g. using jest) or integration tests (e.g. using Playwright/Cypress) - unfortunately, we don't have integration tests set up for the web UI project just yet.

Adopting test-driven development when working on new features also helps you identify issues a lot earlier. You can use the generated code coverage report to see the areas haven't been tested, which can then help you come up with tests for new scenarios (but don't write tests just to cover specific lines of code).

Of course, running the web UI locally and trying the changes out manually also helps you spot issues that you may not have picked up on in the automated tests.

Hope that helps!

shivambisen commented:
Got it, thanks for the tips @eamansour I’ll keep that in mind going forward! 👍
